specification/vmware/resource-manager/Microsoft.AVS/preview/2019-08-09-preview/vmware.json

```diff
@@ -0,0 +1,1277 @@
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/PrivateCloud"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ApiError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PrivateClouds_Get": {
+            "$ref": "./examples/PrivateClouds_Get.json"
+          }
+        }
+      },
+      "put": {
+        "operationId": "PrivateClouds_CreateOrUpdate",
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "PrivateClouds"
+        ],
+        "summary": "Create or update a private cloud",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/privateCloudName"
+          },
+          {
+            "name": "privateCloud",
+            "description": "The private cloud",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PrivateCloud"
+            }
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/PrivateCloud"
+            }
+          },
+          "201": {
+            "description": "accepted operation",
+            "schema": {
+              "$ref": "#/definitions/PrivateCloud"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ApiError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PrivateClouds_CreateOrUpdate": {
+            "$ref": "./examples/PrivateClouds_CreateOrUpdate.json"
+          }
+        }
+      },
+      "patch": {
+        "operationId": "PrivateClouds_Update",
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "PrivateClouds"
+        ],
+        "summary": "Update a private cloud",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/privateCloudName"
+          },
+          {
+            "name": "privateCloud",
+            "description": "The private cloud",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PrivateCloud"
```

Comment:
>PrivateCloud [](start = 37, length = 12)

You are reusing the same definition for patch body as put body. This implies that everything is patchable, which isn't actually true (`location` can't be changed, for example). We recommend creating a sub-set definition for patch body that more accurately represents which properties can be changed with a PATCH request.

Reply:
I will look into addressing this in the 2020-03-20-preview API, but not this one.

Reply:
This is relatively minor, so I will let that slide.

---
In reply to: [419676862](https://github.com/Azure/azure-rest-api-specs/pull/9307#discussion_r419676862) [](ancestors = 419676862)